pallets/cash/src/core.rs

```diff
@@ -429,192 +404,24 @@ pub fn dispatch_extrinsics_internal<T: Config>(extrinsics: Vec<Vec<u8>>) -> Resu
-
-    ChainCashPrincipals::insert(chain_id, chain_cash_principal_new);
-    CashPrincipals::insert(holder, holder_cash_principal_new);
-    TotalCashPrincipal::put(total_cash_principal_new);
-
-    internal::notices::dispatch_cash_extraction_notice::<T>(recipient, principal);
-
-    <Module<T>>::deposit_event(Event::ExtractCash(holder, recipient, principal, index));
-
-    Ok(())
-}
-
-pub fn transfer_internal<T: Config>(
-    asset: AssetInfo,
-    sender: ChainAccount,
-    recipient: ChainAccount,
-    amount: AssetQuantity,
-) -> Result<(), Reason> {
-    let miner = get_some_miner::<T>();
-    let index = GlobalCashIndex::get();
-    let fee_principal = index.cash_principal_amount(TRANSFER_FEE)?;
-
-    require_min_tx_value!(get_value::<T>(amount)?);
-
-    CashPipeline::new()
-        .transfer_asset::<T>(sender, recipient, asset.asset, amount)?
-        .transfer_cash::<T>(sender, miner, fee_principal)?
-        .check_collateralized::<T>(sender)?
-        .commit::<T>();
-
-    <Module<T>>::deposit_event(Event::Transfer(
-        asset.asset,
-        sender,
-        recipient,
-        amount.value,
-    ));
-
-    Ok(())
-}
-
-pub fn transfer_cash_principal_internal<T: Config>(
-    sender: ChainAccount,
-    recipient: ChainAccount,
-    principal: CashPrincipalAmount,
-) -> Result<(), Reason> {
-    let miner = get_some_miner::<T>();
-    let index = GlobalCashIndex::get();
-    let fee_principal = index.cash_principal_amount(TRANSFER_FEE)?;
-    let amount = index.cash_quantity(principal)?;
-
-    require_min_tx_value!(get_value::<T>(amount)?);
-
-    CashPipeline::new()
-        .transfer_cash::<T>(sender, recipient, principal)?
-        .transfer_cash::<T>(sender, miner, fee_principal)?
-        .check_collateralized::<T>(sender)?
-        .commit::<T>();
-
-    <Module<T>>::deposit_event(Event::TransferCash(sender, recipient, principal, index));
-
-    Ok(())
-}
-
-/// Calculates the current CASH principal of the account, including all interest from non-CASH markets.
-pub fn get_cash_principal_with_asset_interest<T: Config>(
-    account: ChainAccount,
-) -> Result<CashPrincipal, Reason> {
-    let mut principal = CashPrincipals::get(account);
-    for (asset, _) in AssetsWithNonZeroBalance::iter_prefix(account) {
-        let asset_info = get_asset::<T>(asset)?;
-        let balance = AssetBalances::get(asset, account);
-        (principal, _) =
-            effect_of_asset_interest_internal(asset_info, account, balance, balance, principal)?;
-    }
-    Ok(principal)
-}
-
 /// Calculates the current total CASH value of the account, including all interest from non-CASH markets.
 pub fn get_cash_balance_with_asset_interest<T: Config>(
     account: ChainAccount,
 ) -> Result<Balance, Reason> {
-    Ok(GlobalCashIndex::get()
-        .cash_balance(get_cash_principal_with_asset_interest::<T>(account)?)?)
-}
+    let cash_index = GlobalCashIndex::get();
```

Comment:
This was already implemented in another place, so just reusing that code to reduce duplication.